### DISCUSSION TITLE
A bunch of questions about pricing, costs and billing

### DISCUSSION THREAD

**Discussion by elmau** [mozilla/hubs#2836] [General Questions & Discussion] [answered]
Hi. I've been trying to find out about the costs involved on hiring hubs-cloud for an event, however aside of the estimates I'm not sure about other costs that can be associated with using the platform.

The event that I'm trying to run has 10 rooms for 25-30 people each, and it has the same music and video streaming to them, also with embedded videos, pdfs, and custom avatars. It's supposed to run for 7 consecutive days, and using the costs estimate chart it seems an event like that could cost around $1600,00 at the worst case scenario. However I'm not sure if that involves the storage space, audio, video streaming and documents.

I thought about using a c5.2xlarge enterprise server for this event. And when I go to the aws site, the price says: $188,37 /month, but I don't know how that price is related to the estimates chart.

Is there any experience of a real case scenario to have an understanding of the costs of using hubs-cloud?

I'm assuming for my event to run at maximum settings 24/7 for a week, and then shut it down. So I guess I shouldn't go over those $1600 that were calculated, but I would like more precision to avoid surprises.

Also I tried to use the cost explorer, but it seems I have to hire the hubs-cloud first to forecast the costs, and my company doesn't want to commit to a contract with AWS if we're not going to use the service.

**Comment by robinkwilson** [comment, marked as answer]
This is an excellent question and a difficult one to answer, I'm hoping other users will share their experiences with costs to give you a better idea. all I can do is give you my rough and tentative estimation method -- so far, I have not heard of being out of the ballpark and I've incorporated what I've found out from the community.

**Disclaimer: Estimating costs is extremely difficult since aws bills by resource usage. Although I can't guarantee a perfect estimate, I'll give you a tentative ballpark and would love to hear how well we did estimating.** Your costs can be minimized further by your diligence with scaling the resources up and down during the live event and setting it to offline mode when no one is connected.

Use aws's cost explorer to use previous costs to estimate future ones.

Hubs Cost Documentation Pages:
[Costs + Minimizing Costs](https://hubs.mozilla.com/docs/hubs-cloud-aws-costs.html)
[Estimating costs](https://hubs.mozilla.com/docs/hubs-cloud-aws-estimated-cost-charts.html)
[Estimating CCU Limits](https://hubs.mozilla.com/docs/hubs-cloud-aws-estimated-ccu-limits.html)

To minimize your costs follow this workflow to manually update the stack + server size during live event: [Minimizing costs user story](https://hubs.mozilla.com/docs/hubs-cloud-aws-costs.html#minimize-your-costs---a-user-story)

### My Rough Model for Estimating

Enterprise multi-server with 2 app x 2 streaming c5.2xlarge servers (4 EC2 instances in total). 

The reason for Enterprise 2x2 multi-server setup is optimizing for resiliency. If one server goes down suddenly, the other will take it's place and your users will not notice.

**Low cost estimate** = # Hours x $/hr Cost for EC2 x # of servers x 2 (scalar low estimate of other services)
**High cost estimate** = # Hours x $/hr Cost for EC2 x # of servers x 5 (scalar high estimate of other services)

1x c5.2xlarge EC2 server = $0.440/hour ([Min. cost on the alpha charts](https://hubs.mozilla.com/docs/hubs-cloud-aws-estimated-cost-charts.html))
Lower bound cost for 4 c5.2xlarge running HC = **Hours at top capacity** x $0.440 x 4 x 2 -- _in the past we've seen EC2 x2 for EC2 + RDS_ 
Higher bound cost for 4 c5.2xlarge running HC = **Hours at top capacity** x $0.440 x 4 x 5

**Bottom line** 
**If you are diligent with decreasing the # of hours at top capacity as mentioned in the [minimizing costs user story guide](https://hubs.mozilla.com/docs/hubs-cloud-aws-costs.html#minimize-your-costs---a-user-story) your event costs can be extremely low especially when comparing an in person event**:
- Scale the EC2 instance down during lower traffic
- Turn on offline mode (costs are extremely minimal because the EC2 costs + RDS costs + EFS costs are down and you're only storing backups)
- Enable database pausing
- Use Cloudflare for content CDN - not recommended if you're streaming videos

Hope this helps. This is the best answer I can provide you. Please review the documentation around this as well.

**Reply by robinkwilson** [reply]
PR for hubs docs update to the Cost Chart page. 

https://github.com/MozillaReality/hubs-docs/pull/64

**Comment by elmau** [comment]
Thank you Robin!, the scalar number on your formula gave us a different price from what we calculated, but still seems manageable for having an event for some hours.

**Reply by Thitle** [reply]
So @elmau would very much like to have your cost feedback on your event, were you able to accurately estimate the costs? and did they reflect on the reality of the event? Thank you!

**Reply by elmau** [reply]
Using the formula for estimating the costs was kind of accurate, but what drove the price up was having some streaming videos in place, some of them being loaded from the same AWS instance. However it wasn't that much off of what we projected for the event. Also we expected a lot more attendance so we warned our client about the possible costs. The event was around $350 for 16 hours (split in two days). With an estimated attendance of 700 people in total. 

Looking at the numbers, I think if the attendance that we expected showed up, we easily could have gone over $2500.

**Reply by Thitle** [reply]
Thanks Elmau for the feedback!!!
